src/Data/LLVM/BitCode/BitString.hs

```diff
@@ -33,35 +40,62 @@ pattern Bytes' n = NumBytes n
 bitCount :: NumBits -> Int
 bitCount (NumBits n) = n
 
+{-# INLINE addBitCounts #-}
 addBitCounts :: NumBits -> NumBits -> NumBits
 addBitCounts (NumBits a) (NumBits b) = NumBits $ a + b
 
+{-# INLINE subtractBitCounts #-}
 subtractBitCounts :: NumBits -> NumBits -> NumBits
 subtractBitCounts (NumBits a) (NumBits b) = NumBits $ a - b
 
+{-# INLINE bytesToBits #-}
+bitsToBytes :: NumBits -> (NumBytes, NumBits)
+bitsToBytes (NumBits n) = (NumBytes $ n `shiftR` 3, NumBits $ n .&. 7)
```

Comment:
Presumably, the use of `shiftR` here will give strange results if `n` is negative? I'm guessing that there is an invariant that `NumBits` is non-negative. If so, it would be worth commenting that near `NumBits`' definition. (Similarly for `NumBytes`.)

Reply:
It would cause issues, and although this isn't any different that the previous code it's reasonable to enforce some invariants around this (or even change the core types).  I'll add these in a future PR (not yet written) if that's OK.